subiquity/server/controllers/install.py

```diff
@@ -130,12 +162,92 @@ async def configure_apt(self, *, context):
         configurer = await mirror.wait_config()
         return await configurer.configure_for_install(context)
 
+    async def run_curtin_install_step(
+            self, step: CurtinInstallStep,
+            resume_data_file: Path,
+            context, source) -> subprocess.CompletedProcess:
+        """ Run a curtin install step. """
+
+        self.app.note_file_for_apport(
+                f"Curtin{step.name}Config", str(step.config_file))
+
+        self.write_config(
+                config_file=step.config_file,
+                config=step.acquire_config(step, resume_data_file)
+                )
+
+        # Make sure the log directory exists.
+        step.log_file.parent.mkdir(parents=True, exist_ok=True)
+
+        # Add a marker to identify the step in the log file.
+        with open(str(step.log_file), mode="a") as fh:
+            fh.write(f"\n---- [[ subiquity step {step.name} ]] ----\n")
+
+        return await run_curtin_command(
+                self.app, context,
+                "install", source,
+                "--set", f'json:stages={json.dumps(step.stages)}',
+                config=str(step.config_file),
+                private_mounts=False)
+
     @with_context(
         description="installing system", level="INFO", childlevel="DEBUG")
     async def curtin_install(self, *, context, source):
-        await run_curtin_command(self.app, context, 'install', source,
-                                 config=self.write_config(),
-                                 private_mounts=False)
+        logs_dir_prefix = Path(
+                self.app.opts.output_base if self.app.opts.dry_run else "/")
+
+        logs_dir = logs_dir_prefix / "var/log/installer"
+        logs_dir.mkdir(parents=True, exist_ok=True)
+
+        config_dir = logs_dir / "curtin-install"
+
+        install_log_file = logs_dir / "curtin-install.log"
```

Comment:
I'm curious why you chose to have all logs go to the same file?

Reply:
He started with one log per stage, and I asked that we unify them.  I thought it would be easier for administrators to diagnose issues, instead of having to think about it in the 'stages'.

Reply:
Fair enough :)